src/consumer_queue.cpp

```diff
@@ -1,13 +1,23 @@
 #include "consumer_queue.h"
 #include "common.h"
-#include "sample.h"
 #include "send_buffer.h"
 #include <chrono>
 
 using namespace lsl;
 
-consumer_queue::consumer_queue(std::size_t max_capacity, send_buffer_p registry)
-	: registry_(registry), buffer_(max_capacity) {
+consumer_queue::consumer_queue(std::size_t size, send_buffer_p registry)
```

Comment:
`std::uint_fast32_t` would let the compiler choose the best int size that's at least 4 bytes wide

Reply:
Oh interesting, didn't know about that one! On that, I've actually been wondering if we want to keep limiting our buffers to 32-bit sizes in the long run (e.g., if I declare a stream with a 6 MHz sampling rate, and if someone connects to that with a default 360 s max_buflen (e.g. LabRecorder), that will crash liblsl currently with no error message since max_buflen (being an int) will wrap around to a negative value that then blows up later). That happened to me while testing jfrey's use case. It made me think that we might want to, in the very long run, replace some things that are currently `int` by `size_t`, and one could slowly but surely build up towards without breaking anything by starting with some low-level containers. The external API wouldn't be affected by that since the size value there is before it's multiplied by the potentially huge sampling rate. 

Just something to think about, especially now that 6 MHz is actually achievable with a sufficiently speedy PC, and also considering that some specialized/professional use cases may find the ability to have such large buffers practical.

Reply:
A sample is at least 48 bytes on x64 so that'd require at least 192GB RAM per queue. Not totally impossible, but I not the typical LSL users.